### PR TITLE
[CI] Run playwright tests with images from CI build using minikube

### PR DESCRIPTION
**Pull request by jankalinic**
This pull request introduces a workflow to set up and validate a Minikube environment with Kafka cluster, Prometheus instance, Strimzi operator, and Console. It should replace Playwright IT tests.

Note: 
* Need some smart way to deploy console-datagen container so it produces data.
* PW tests are unable to reach any webpage, but curl check works (returns kafka Id).

### CODE REVIEW

**Comment by jankalinic** [comment]
So workflow seems working (even with curl returning kafkaID from the running kafka cluster). I may have a questions coming from my notes, that i would like to discuss with you @streamshub/console-developers 

How can i effectively deploy console-datagen  in this case, so it generates test data ?
What else do i need to do to run the playwright tests against this setup? (they are crashing on the build part, but previously they could not connect to any of the pages, so they might require further changes in settings)

Thanks all for your help

**Comment by MikeEdgar** [comment]
> How can i effectively deploy console-datagen in this case, so it generates test data ?

Something like this should work. You may need to adjust the `DATAGEN_` envs depending on how much CPU you have to spare. Note there is also a `${CLUSTER_DOMAIN}` value that needs to be replaced with the Kube domain like you've done with the `Console` CR instance.

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: console-datagen
spec:
  replicas: 1
  revisionHistoryLimit: 3
  selector:
    matchLabels:
      app: console-datagen
  template:
    metadata:
      labels:
        app: console-datagen
    spec:
      containers:
      - name: console-datagen
        image: quay.io/streamshub/console-datagen
        ports:
        - containerPort: 9080
        livenessProbe:
          httpGet:
            path: /q/health/live
            port: 9080
            scheme: HTTP
          initialDelaySeconds: 10
          periodSeconds: 10
          successThreshold: 1
          failureThreshold: 3
          timeoutSeconds: 10
        env:
        - name: KAFKA_SECURITY_PROTOCOL
          value: SASL_SSL
        - name: KAFKA_SASL_MECHANISM
          value: SCRAM-SHA-512
        - name: DATAGEN_CONSUMER_GROUPS
          value: "3"
        - name: DATAGEN_TOPICS_PER_CONSUMER
          value: "2"
        - name: DATAGEN_PARTITIONS_PER_TOPIC
          value: "3"
        - name: DATAGEN_KAFKA_TEST_NAME
          value: "TestKafka"
        - name: DATAGEN_KAFKA_TEST_CONFIGS_BOOTSTRAP_SERVERS
          value: "bootstrap.console-kafka.${CLUSTER_DOMAIN}:443"
        - name: DATAGEN_KAFKA_TEST_CONFIGS_SASL_JAAS_CONFIG
          valueFrom:
            secretKeyRef:
              key: sasl.jaas.config
              name: console-kafka-user1
```

**Comment by jankalinic** [comment]
Thanks Michael ! So all i need to know now is how to setup the playwright to run tests successfully, as they will result in the failures described earlier. @riccardo-forina Can you guide me with this one please?

**Comment by jankalinic** [comment]
@riccardo-forina do i need to change anything else for the playwright to start working with the minikube instead of the localhost js server?

File: .github/workflows/minikube-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      - name: Run Playwright Tests
+        env:
+          BACKEND_URL: ${{ secrets.BACKEND_URL }}
+          CONSOLE_METRICS_PROMETHEUS_URL: ${{ secrets.CONSOLE_METRICS_PROMETHEUS_URL }}
+          NEXTAUTH_SECRET: ${{ secrets.NEXTAUTH_SECRET }}
+          LOG_LEVEL: ${{ secrets.LOG_LEVEL }}
+          CONSOLE_MODE: ${{ secrets.CONSOLE_MODE }}
+          HOME: /root
+        run: |
+          npm build
```
Comment (MikeEdgar) [inline .github/workflows/minikube-deployment.yml:138]:
```suggestion
          npm run build
```

**Review by MikeEdgar — COMMENTED** [review]
This seems close... there's one typo that may be all that's needed to fix it.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -156,4 +171,10 @@ jobs:
           provenance: false
           push: false
           tags: |
-            streamshub/console-ui:${{ env.PROJECT_VERSION }}
+            localhost:5000/streamshub/console-ui:${{ env.PROJECT_VERSION }}
+
+  call-minikube-deployment:
```
Comment (MikeEdgar) [inline .github/workflows/integration.yml:176]:
```suggestion
  playwright-tests:
```

File: .github/workflows/minikube-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
+name: Minikube Deployment
```
Comment (MikeEdgar) [inline .github/workflows/minikube-deployment.yml:1]:
```suggestion
name: Playwright Tests
```

File: .github/workflows/minikube-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  workflow_call:
+
+env:
+  TARGET_NAMESPACE: "console-namespace"
+  CI_CLUSTER: true
+  CONSOLE_INSTANCE_YAML: "./install/resources/console/console.instance.yaml"
+  CLUSTER_DOMAIN: "192.168.49.2.nip.io"
+
+jobs:
+  setup_deployment:
```
Comment (MikeEdgar) [inline .github/workflows/minikube-deployment.yml:13]:
```suggestion
  tests:
```

**Review by MikeEdgar — APPROVED** [review]

**Comment by jankalinic** [comment]
@MikeEdgar Thank you so much for your work! It looks awesome. I guess we can merge it.

**Review by riccardo-forina — APPROVED** [review]

**Pull request merged**